pkg/manager/internal.go

```diff
@@ -357,6 +359,18 @@ func (cm *controllerManager) GetWebhookServer() *webhook.Server {
 	return cm.webhookServer
 }
 
+func (cm *controllerManager) GetRESTClientFor(obj runtime.Object) (rest.Interface, error) {
+	gvk, err := apiutil.GVKForObject(obj, cm.GetScheme())
+	if err != nil {
+		return nil, err
+	}
+	restClient, err := apiutil.RESTClientForGVK(gvk, cm.GetConfig(), serializer.NewCodecFactory(cm.GetScheme()))
+	if err != nil {
+		return nil, err
+	}
+	return restClient, nil
+}
```

Comment:
Should this be here, or rather have a function in `controllerutil` where you could pass the Scheme and object in?